omegaconf/listconfig.py

```diff
@@ -256,7 +255,7 @@ def append(self, item: Any) -> None:
                 ref_type=self.__dict__["_metadata"].element_type,
                 key=index,
                 value=item,
-                is_optional=_is_optional(self),
+                is_optional=item._is_optional() if isinstance(item, Node) else True,
```

Comment:
and now it becomes clear why OmegaConf.is_optional returned True for none Nodes.
I think we can restore this behavior (but document it).

Reply:
Done in https://github.com/omry/omegaconf/pull/705/commits/bc3b7a6239be80164b018ffd9251e1eeeebcacd4.